File: CS/AppointmentRecurrenceFormExample/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using DevExpress.XtraScheduler;
using DevExpress.Web.Mvc;
using DevExpress.XtraScheduler.Native;

namespace AppointmentRecurrenceFormExample.Controllers
{
    public class HomeController: Controller {
        public ActionResult Index() {
            RecurrenceInfo recurrenceInfo = new RecurrenceInfo() {
                Start = DateTime.Today,
                End = DateTime.Today.Add(TimeSpan.FromDays(1)),
                Type = RecurrenceType.Weekly,
                AllDay = true
            };
            return View("Index", recurrenceInfo);
        }
        public ActionResult RecurrenceFormInfo() {
            Appointment pattern = DevExpress.XtraScheduler.Compatibility.StaticAppointmentFactory.CreateAppointment(AppointmentType.Pattern);
            pattern.Start = new DateTime(2012, 10, 20, 3, 0, 0);
            pattern.End = new DateTime(2012, 10, 20, 4, 0, 0);
            RecurrenceInfo recurrenceInfo = AppointmentRecurrenceFormExtension.GetValue("RecurrenceControl", pattern);
            return View("RecurrenceFormInfo", (object)recurrenceInfo.ToXml());
        }
    }
}

File: VB/AppointmentRecurrenceFormExample/Controllers/HomeController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc
Imports DevExpress.XtraScheduler
Imports DevExpress.Web.Mvc

Namespace AppointmentRecurrenceFormExample.Controllers
	Public Class HomeController
		Inherits Controller
		Public Function Index() As ActionResult
			Dim recurrenceInfo As New RecurrenceInfo() With {.Start = DateTime.Today, .End = DateTime.Today.Add(TimeSpan.FromDays(1)), .Type = RecurrenceType.Weekly, .AllDay = True}
			Return View("Index", recurrenceInfo)
		End Function
		Public Function RecurrenceFormInfo() As ActionResult
			Dim pattern As New Appointment(AppointmentType.Pattern, New DateTime(2012, 10, 20, 3, 0, 0), New DateTime(2012, 10, 20, 4, 0, 0))
			Dim recurrenceInfo As RecurrenceInfo = AppointmentRecurrenceFormExtension.GetValue("RecurrenceControl", pattern)
			Return View("RecurrenceFormInfo", CObj(recurrenceInfo.ToXml()))
		End Function
	End Class
End Namespace
